CMakeLists.txt

```diff
@@ -85,6 +85,9 @@ mongo_setting(ENABLE_ZSTD "Enable zstd compression support"
 mongo_setting(ENABLE_ZLIB "Enable zlib compression support"
               OPTIONS BUNDLED SYSTEM OFF
               DEFAULT VALUE BUNDLED)
+mongo_setting(ENABLE_UTF8PROC "Enable building with utf8proc. Needed for SCRAM-SHA-256 authentication with non-ASCII passwords"
+              OPTIONS BUNDLED SYSTEM
+              DEFAULT VALUE BUNDLED)
```

Comment:
These may be questions for @rcsanchez97 and @vector-of-bool.

This PR bundles `utf8proc` to help remove the dependency on ICU.

- If `utf8proc` is bundled, is an option needed to use a system install of `utf8proc`?
- Would Debian packaging require an option to use a system install of `utf8proc`?

IMO `utf8proc` is a small library. I prefer not having another option between using a system install or the bundled if it is not needed. `utf8proc` appears to be packaged on many systems (https://repology.org/project/utf8proc/versions).

Using a `SYSTEM` install of IntelDFP in libmongocrypt was added (required?) for Debian packaging in MONGOCRYPT-517. If supporting a system install of `utf8proc` is necessary, more work may be needed in this PR to support the `SYSTEM` value.

Reply:
So, the short answer is "yes, Debian (and Fedora) will require the ability to build against a system-provided utf8proc."

Additionally, from a Debian packaging perspective, we will want to filter out the embedded code (as we do for zlib, which can be seen in `debian/gdb.conf`). In essence, embedding code copies is very strongly discouraged in Debian (Fedora discourages it as well, but they aren't quite as stringent about it). We would have to have a very good reason for the archive maintainers to OK the introduction of an embedded code copy. We probably don't have a very good reason here, since the utf8proc package is available at the current latest upstream version.

The Debian-specific parts of this could be a little tricky. So, my recommendation is that Josh handle implementing the ability to specify a system install of utf8proc to the CMake configuration, and then I can handle updating the Debian packaging parts in a follow-up PR.

Reply:
I would recommend changing the option to be a boolean `USE_BUNDLED_UTF8PROC` (default `TRUE`), and mark it as `ADVANCED`. Because utf8proc does not (appear to) ship a CMake config-module, we can add our own find-module `Findutf8proc.cmake` that imports the external library. We will also require modification to the `mongoc-1.0-config.cmake` to also handle the case where mongoc was built with `USE_BUNDLED_UTF8PROC`.